props/saber_fett263_buttons.h

```diff
@@ -293,248 +659,3857 @@ class SaberFett263Buttons : public virtual PropBase {
+      case MENU_FORCEPUSH:
+        choice_ = false;
+        PlayMenuSound("mdisable.wav");
+        break;         
+      case MENU_FORCEPUSH_LENGTH:
+      case MENU_CLASH_DETECT:
+        if (calc_ > 1) {
+          PlayMenuSound("mdown.wav");
+          calc_ -= 1;
+        }
+        if (calc_ <= 1) {
+          calc_ = 1;
+          PlayMenuSound("mmin.wav");
+        }
+        SayNumber(calc_, SAY_WHOLE);
+        break;
+      case MENU_MAX_CLASH:
+        if (calc_ > 8) {
+          PlayMenuSound("mdown.wav");
+          calc_ -= 1;
+        }
+        if (calc_ <= 8) {
+          calc_ = 8;
+          PlayMenuSound("mmin.wav");
+        }
+        SayNumber(calc_, SAY_WHOLE);
+        break;
+      case MENU_TWISTOFF:
+        choice_ = false;
+        PlayMenuSound("mdisable.wav");
+        break;         
+      case MENU_LOCKUP_DELAY:
+        if (calc_ > 200) {
+          PlayMenuSound("mdown.wav");
+          calc_ -= 100;
+        }
+        if (calc_ <= 200) {
+          calc_ = 200;
+          PlayMenuSound("mmin.wav");
+        }
+        SayNumber(calc_, SAY_WHOLE); 
+        break;
+      case MENU_POWERLOCK:
+        choice_ = false;
+        PlayMenuSound("mdisable.wav");
+        break;         
+      case MENU_SETTING_SUB:
+        if (menu_sub_pos_ <= 1) menu_sub_pos_ = SET_SUBMENUS + 1;
+        menu_sub_pos_ -= 1;
+        switch (menu_sub_pos_) {
+          case 1:
+            PlayMenuSound("mvolume.wav");
+            break;
+          case 2:
+            PlayMenuSound("mstylset.wav");
+            break;
+          case 3:
+            PlayMenuSound("mcontrol.wav");
+            break;
+          case 4:
+            PlayMenuSound("mclashth.wav");
+            break;
+          case 5:
+            PlayMenuSound("mlength.wav");
+            break;
+          case 6:
+            PlayMenuSound("mdim.wav");
+            break;
+          default:
+            break;
+        }
+        break;
+      case MENU_CLASH_THRESHOLD:
+        if (clash_t_ <= 1.0) {
+          PlayMenuSound("mmin.wav");
+          clash_t_ = 1.0;
+        } else {
+          PlayMenuSound("mdown.wav");
+          clash_t_ -= 0.25;
+          if (clash_t_ < 1.0) clash_t_ = 1.0;
+        }
+        SayNumber(clash_t_, SAY_DECIMAL);
+        break;
+      case MENU_DIM_BLADE:
+        dim = std::max<float>(dim - 0.1, 0.2);
+        if (dim == 0.2) {
+          PlayMenuSound("mmin.wav");
+        } else {
+          PlayMenuSound("mdown.wav");
+        }
+        SaberBase::SetDimming(pow(dim, 2.2) * 16384);
+        break;
+      case MENU_STYLE_SETTING_SUB:
+        while (true) {
+          if (effect_num_ == 0) effect_num_ = STYLE_SETTINGS + 1;
+          effect_num_ -= 1;
+          if (style_parser.UsesArgument(current_preset_.GetStyle(blade_num_), effect_num_ + 17)) break;
+        }
+        set_num_ = effect_num_ + 15;
+        switch (set_num_) {
+          case 16:
+            PlayMenuSound("mstylopt.wav");
```

Comment:
We're going to need to make a class that has a list of all the menu sounds, the class could look like this:

class MenuSoundHelper {
   public:
   virtual PlayMenuSound(const char* str) = 0;
   void playStyleOption() { PlayMenuSound("mstyleopt.wav"); } 
   void playIgnitionOption() { PlayMenuSound("migopt.wav"); }    
   ....
};

The list of functions and their names would serve as documentation for what sounds are available when over people wants to use them.

In this class, we just inherit the helper class and instead of PlayMenuSound("mstyleopt.wav") we call playStyleOption(); 

Note that I guessed on the function names. They should match what the voice says as closely as possible. So if the voice says "style option menu" the name should be playStyleOptionMenu(). If the voice says "style options", then the name should be playStyleOptions(), etc.

Reply:
Is this necessary, looking through and it's going to be pretty time-consuming?  My documentation already has all prompt filenames and spoken words listed and the /common folder provided by Brian will already contain all sounds necessary for users to implement.  I can put a copy of the documentation in the comments at top of file as a duplicate but replacing all prompts with new functions seems like overkill.

Reply:
Yes, we do need to do this, but we could do it after the initial check-in.